produce/lib/produce/merchant.rb

```diff
@@ -0,0 +1,83 @@
+require 'spaceship'
+
+module Produce
+  class Merchant
+    def create(_options, _args)
+      login
+
+      merchant_identifier = Produce.config[:merchant_identifier] || UI.input("Merchant identifier (reverse-domain name style string starting with 'merchant'): ")
+
+      if merchant_exists? merchant_identifier
+        UI.success("[DevCenter] Merchant '#{merchant_identifier})' already exists, nothing to do on the Dev Center")
+      else
+        merchant_name = Produce.config[:merchant_name] || merchant_identifier.split(".").map(&:capitalize).reverse.join(' ')
+
+        UI.success("Creating new merchant '#{merchant_name}' with identifier '#{merchant_identifier}' on the Apple Dev Center")
+
+        merchant = Spaceship.merchant.create!(bundle_id: merchant_identifier,
+                                            name: merchant_name,
+                                            mac: Produce.config[:platform].to_s == 'mac')
+
+        if merchant.name != merchant_name
+          UI.important("Your merchant name includes non-ASCII characters, which are not supported by the Apple Developer Portal.")
+          UI.important("To fix this a unique (internal) name '#{merchant.name}' has been created for you.")
+        end
+
+        UI.message("Created merchant #{merchant.merchant_id}")
+        UI.user_error!("Something went wrong when creating the new merchant - it's not listed in the merchants list") unless merchant_exists? merchant_identifier
+        UI.success("Finished creating new merchant '#{merchant_name}' on the Dev Center")
```

Comment:
Can we reorder these so that the user error comes first (and also wrap `merchant_identifier` as the parameter to `merchant_exists?` in parentheses) and then the success messages, it looks like this will always say that it was created even though the next line says it failed (if I understand this all correctly?)